File: WikiBeer/Model/Ingredients/Ingredient.cs
using Ipme.WikiBeer.Model.Magic;
using System;
using System.Collections.Generic;
using System.Text;

/// <summary>
/// TODO : 
/// </summary>
namespace Ipme.WikiBeer.Model.Ingredients
{
    public abstract class Ingredient
    {
        public Guid Id { get; }
        public string Name { get; internal set; }
        public string Description { get; internal set; }
        public string TypeOfIngredient { get; private set; } // Pour test pour l'instant 

        public Ingredient(string name, string desription = Rules.DEFAULT_INGREDIENT_DESCRIPTION)
        {
            // Définitif
            Id = Guid.NewGuid();
            Name = name;            
            Description = desription;

            // Pour test
            TypeOfIngredient = this.ToString(); 
        }
    }
}

File: WikiBeer/Web/src/models/ingredients/ingredient.ts
export abstract class Ingredient{

  protected _id: string = '';
  protected _name: string = '';
  protected _description: string ='';

  protected constructor(id: string, name: string, description: string)
  {
    this._id = id;
    this._name = name;
    this._description = description;
  }

  get id(): string {return this._id;}
  set id(value: string) {this._id = value;}

  get name(): string {return this._name;}
  set name(value: string) {this._name = value;}

  get description(): string {return this._description;}
  set description(value: string) {this._description = value;}

}
